_studio/mfx_lib/decode/vp8/src/mfx_vp8_dec_decode_common.cpp

```diff
@@ -258,20 +257,21 @@ mfxStatus MFX_VP8_Utility::Query(VideoCORE *p_core, mfxVideoParam *p_in, mfxVide
 
         if (opaque_in && opaque_out)
         {
+            MFX_CHECK(opaque_out->In.Surfaces && opaque_in->In.Surfaces, MFX_ERR_UNDEFINED_BEHAVIOR);
+
             opaque_out->In.Type = opaque_in->In.Type;
             opaque_out->In.NumSurface = opaque_in->In.NumSurface;
-            MFX_INTERNAL_CPY(opaque_out->In.Surfaces, opaque_in->In.Surfaces, opaque_in->In.NumSurface);
+            std::copy_n(opaque_in->In.Surfaces, opaque_in->In.NumSurface, opaque_out->In.Surfaces);
```

Comment:
It's frequent case when "in" == "out" are passed to Query. "Overlap of ranges is not permitted" for std::copy_n. Let's add check for "opaque_in->In.Surfaces != opaque_out->In.Surfaces" and avoid copy if pointers are same.

Same for opaque_out->Out.Surfaces && opaque_in->Out.Surfaces.

Reply:
done